src/megachatapi_impl.cpp

```diff
@@ -2747,12 +2747,7 @@ const char *MegaChatApiImpl::getUserFirstnameFromCache(MegaChatHandle userhandle
     SdkMutexGuard g(sdkMutex);
     if (mClient && mClient->isUserAttrCacheReady())
     {
-        const Buffer* buffer = mClient->userAttrCache().getDataFromCache(userhandle, ::mega::MegaApi::USER_ATTR_FIRSTNAME);
-        if (buffer != nullptr)
-        {
-            std::string name(buffer->buf(), buffer->size());
-            return MegaApi::strdup(name.c_str());
-        }
+        return move(mClient->userAttrCache().getDataFromCache(userhandle, ::mega::MegaApi::USER_ATTR_FIRSTNAME)->c_str());
```

Comment:
I think `move` method isn't necessary, we have a raw pointer and we are returning it

Reply:
Agree, change it, please.